From 3752b6403d7c1a99feac89f979ff09b0ef84a5ca Mon Sep 17 00:00:00 2001
From: Vinay Verma <vvinay@codeaurora.org>
Date: Fri, 14 Sep 2018 11:25:46 +0530
Subject: [PATCH 18/18] Reduce hwbinder buffer size for power HAL

Set hwbinder buffer size to 16KB for 32 bit architecture.

Change-Id: I4272a12853269dd6363f58a5bb81caa09cc47e9a
CRs-Fixed: 2315283
---
 Android.mk  | 6 ++++++
 service.cpp | 6 ++++++
 2 files changed, 12 insertions(+)

diff --git a/Android.mk b/Android.mk
index 30dcf00..6d7930b 100644
--- a/Android.mk
+++ b/Android.mk
@@ -30,6 +30,7 @@ LOCAL_SHARED_LIBRARIES := \
     libhidlbase \
     libhidltransport \
     libhardware \
+    libhwbinder \
     libutils
 
 LOCAL_SRC_FILES := \
@@ -175,6 +176,11 @@ endif
 ifeq ($(TARGET_HAS_NO_WLAN_STATS),true)
 LOCAL_CFLAGS += -DNO_WLAN_STATS
 endif
+
+ifeq ($(TARGET_ARCH),arm)
+LOCAL_CFLAGS += -DARCH_ARM_32
+endif
+
 LOCAL_MODULE := android.hardware.power@1.1-service-qti
 LOCAL_INIT_RC := android.hardware.power@1.1-service-qti.rc
 LOCAL_SHARED_LIBRARIES += android.hardware.power@1.1 vendor.lineage.power@1.0
diff --git a/service.cpp b/service.cpp
index 4b74156..d43b652 100644
--- a/service.cpp
+++ b/service.cpp
@@ -22,6 +22,9 @@
 #include <android/log.h>
 #include <hidl/HidlTransportSupport.h>
 #include <hardware/power.h>
+#ifdef ARCH_ARM_32
+#include <hwbinder/ProcessState.h>
+#endif
 #include "Power.h"
 
 using android::sp;
@@ -36,6 +39,9 @@ using android::hardware::joinRpcThreadpool;
 using android::hardware::power::V1_1::implementation::Power;
 
 int main() {
+#ifdef ARCH_ARM_32
+    android::hardware::ProcessState::initWithMmapSize((size_t)16384);
+#endif
 
     status_t status;
     android::sp<Power> service = nullptr;
-- 
2.17.1

